[MIG][17.0] edi_exchange_template_oca: Migration to 17.0

john-herholz-dt opened the pull request.

simahawk committed:
Add edi_exchange_template

simahawk committed:
edi_exchange_template: minor changes

oca-travis committed:
[UPD] Update edi_exchange_template.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
edi_exchange_template 13.0.1.1.0

simahawk committed:
edi.backend: add cron to handle record state updates

simahawk committed:
edi.exchange.template: update according to edi changes

OCA-git-bot committed:
edi_exchange_template 13.0.1.2.0

etobella committed:
[FIX] edi_exchange_template Tests

OCA-git-bot committed:
edi_exchange_template 13.0.1.3.0

simahawk committed:
edi.template: fix info provider lookup

simahawk committed:
edi_exchange_template: fix tests class

OCA-git-bot committed:
edi_exchange_template 13.0.1.4.0

simahawk committed:
edi.backend: rename generate to match others

All the other main methods start with 'exchange_'.
Make generate respect this convention.

simahawk committed:
edi_exchange_template: add TODO

simahawk committed:
edi_exchange_template: adapt lookup to new api

OCA-git-bot committed:
edi_exchange_template 13.0.1.5.0

etobella committed:
[CHG] edi: Use more permissive licence: AGPL-> LGPL

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
edi_exchange_template 13.0.1.5.1

OCA-git-bot committed:
edi_exchange_template 13.0.1.6.0

etobella committed:
[IMP] edi_exchange_template: Allow to use report instead of view

etobella committed:
[IMP] edi_exchange_template_oca: black, isort, prettier

etobella committed:
[MIG] edi_exchange_template_oca: Migration to 14.0

oca-travis committed:
[UPD] Update edi_exchange_template_oca.pot

OCA-git-bot committed:
[UPD] README.rst

pedrobaeza committed:
[IMP] edi_exchange_template_oca: Promote to beta

In use for several modules.

OCA-git-bot committed:
[UPD] README.rst

pedrobaeza committed:
edi_exchange_template_oca 14.0.1.0.1

JuaniFreedoo committed:
[MIG] edi_change_template_oca : Migration to 15.0

Commit:
[UPD] Update edi_exchange_template_oca.pot

OCA-git-bot committed:
[UPD] README.rst

etobella committed:
[IMP] edi_exchange_template: Allow users to see, but not edit

OCA-git-bot committed:
edi_exchange_template_oca 15.0.1.0.1

simahawk committed:
edi_exchange_template_oca: fix xml_purge_nswrapper

Use `name()` because the real element could be namespaced on render
leading to mismatch by element type.

simahawk committed:
edi_exchange_template_oca: add nswrapper test

simahawk committed:
edi_exchange_template_oca: update ownership

simahawk committed:
edi_exchange_template: fix component lookup

simahawk committed:
edi_exchange_template: move menu under config

simahawk committed:
edi_exchange_template: improve time utils on render ctx

simahawk committed:
edi_exchange_template: add first_of helper

simahawk committed:
edi_exchange_template: add prettify option

simahawk committed:
edi_exchange_template: drop useless logger

simahawk committed:
edi_exchange_template: fix _get_render_values

Custom kw args for rendering should be injected before snippets are evaluated.
This way they are available for the snippet
but also the snippet can override any custom param.

simahawk committed:
edi_exchange_template: fix output mixin name

simahawk committed:
edi_exchange_template: fix mixin inherit

simahawk committed:
edi_exchange_template: partial fix for ctx propagation

OCA-git-bot committed:
[UPD] README.rst

etobella committed:
[IMP] edi_excange_template_oca: black, isort, prettier

etobella committed:
[MIG] edi_exchange_template_oca: Migration to 16.0

Commit:
[UPD] Update edi_exchange_template_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

simahawk committed:
edi_exchange_template: adapt tests

exchange_generate now returns status string, not the output.

Commit:
[IMP] edi_exchange_template_oca: pre-commit

Commit:
[UPD] Update edi_exchange_template_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

victoralmau committed:
[FIX] edi_exchange_template_oca: Change SavepointCase to TransactionCase

OCA-git-bot committed:
[BOT] post-merge updates

peluko00 commented:
Can you push again @john-herholz-dt please

john-herholz-dt commented:
I did but there is still an unmet dependency

peluko00 commented:
> I did but there is still an unmet dependency

It's normal because depends of other pr. Thanks!

peluko00 reviewed (approved):
LGTM, code review

john-herholz-dt committed:
[IMP] pre-commit auto fixes

john-herholz-dt committed:
[MIG] edi_exchange_template_oca: Migration to 17.0

peluko00 commented:
Test are failing @john-herholz-dt

manuelregidor commented on edi_exchange_template_oca/tests/test_edi_backend_output.py:
```diff
@@ -0,0 +1,178 @@
+            file_content.strip(),
+        )
+
+    def test_get_info_provider(self):
```
This test makes no checks.

john-herholz-dt commented:
I removed the wrong test and added two tiny tests for the date functions to get more coverage.
Some coverage is missing but I think testing about a placeholder function like "validate: pass" does not make much sense.

manuelregidor reviewed (approved):
Technical review. LGTM

HaraldPanten reviewed (approved):
@etobella WDYT? This PR is the other missing dependency for l10n_es_facturae_face.

THX!

etobella reviewed (approved):
/ocabot merge nobump

etobella commented:
/ocabot migration edi_exchange_template_oca
